src/flint/types/fmpz_mod_poly.pyx

```diff
@@ -0,0 +1,1544 @@
+
+        res = fmpz_mod.__new__(fmpz_mod)
+        res.ctx = self.ctx.mod
+        fmpz_mod_poly_resultant(
+            res.val, self.val, (<fmpz_mod_poly>other).val, self.ctx.mod.val
+        )
+        return res
+
+    def sqrt(self):
+        """
+        If ``self`` is a perfect square, compute the square root
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,1])
+            >>> (f*f).sqrt()
+            x + 1
+
+        """
+        cdef fmpz_mod_poly res
+        cdef int check
+
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+        check = fmpz_mod_poly_sqrt(
+            res.val, self.val, res.ctx.mod.val
+        )
+        if check != 1:
+            raise ValueError(
+                f"Cannot compute square-root {self}"
+            )
+        return res
+
+    def sqrt_trunc(self, slong n):
+        """
+        Returns the squareroot of ``self`` modulo `x^n`.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> x = R.gen()
+            >>> f = R([1,2,3])
+            >>> h = f.sqrt_trunc(5)
+            >>> h
+            82*x^4 + 162*x^3 + x^2 + x + 1
+            >>> h.mulmod(h, x**5) == f
+            True
+
+        """
+        cdef fmpz_mod_poly res
+
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+        fmpz_mod_poly_sqrt_series(
+            res.val, self.val, n, res.ctx.mod.val
+        )
+        return res
+
+    def inverse_sqrt_trunc(self, slong n):
+        """
+        Returns the inverse squareroot of ``self`` modulo `x^n`.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,2,3])
+            >>> h = f.inverse_sqrt_trunc(5)
+            >>> h
+            78*x^4 + 2*x^3 + 162*x + 1
+            >>> (h*h).inverse_series_trunc(5) == f
+            True
+        """
+        cdef fmpz_mod_poly res
+
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+        fmpz_mod_poly_invsqrt_series(
+            res.val, self.val, n, res.ctx.mod.val
+        )
+        return res
+
+    def inflation(self, ulong n):
+        r"""
+        Returns the result of the polynomial `f = \textrm{self}` to
+        `f(x^n)`
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,2,3])
+            >>> f.inflation(10)
+            3*x^20 + 2*x^10 + 1
+
+        """
+        cdef fmpz_mod_poly res
+
+        res = fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+        fmpz_mod_poly_inflate(
+            res.val, self.val, n, res.ctx.mod.val
+        )
+        return res
+
+    def deflation(self, ulong n):
+        r"""
+        Returns the result of the polynomial `f = \textrm{self}` to
+        `f(x^{1/n})`
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,0,2,0,3])
+            >>> f
+            3*x^4 + 2*x^2 + 1
+            >>> f.deflation(2)
+            3*x^2 + 2*x + 1
```

Comment:
This is not consistent with the other types:
```python
In [4]: p = flint.fmpz_poly([1, 0, 3])

In [5]: p
Out[5]: 3*x^2 + 1

In [6]: p.deflation()
Out[6]: (3*x + 1, 2)

In [7]: p.deflation(2)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 p.deflation(2)

TypeError: deflation() takes exactly 0 positional arguments (1 given)
```
I think that flint defines deflate and deflation differently.

Reply:
I just realised this. I have pushed a new version where I have both `delate` and `deflation`.